EndlessClient/Dialogs/BarberDialog.cs

```diff
@@ -0,0 +1,223 @@
+            var newHairStyle = (currentProperties.HairStyle + offset) % 21;
+            _characterControl.UpdateRenderProperties(newHairStyle, currentProperties.HairColor,
+                                                     currentProperties.Race, currentProperties.Gender,
+                                                     currentProperties.ArmorGraphic, currentProperties.WeaponGraphic,
+                                                     currentProperties.BootsGraphic, currentProperties.ShieldGraphic);
+            _changeHairItem.SubText = GetCurrentHairStyleText(newHairStyle);
+        }
+
+        private void UpdateHairColor(int offset)
+        {
+            var currentProperties = _characterControl.RenderProperties;
+            var newHairColor = (currentProperties.HairColor + offset) % _hairColorNames.Length;
+            _characterControl.UpdateRenderProperties(currentProperties.HairStyle, newHairColor,
+                                                     currentProperties.Race, currentProperties.Gender,
+                                                     currentProperties.ArmorGraphic, currentProperties.WeaponGraphic,
+                                                     currentProperties.BootsGraphic, currentProperties.ShieldGraphic);
+            _changeHairColor.SubText = GetCurrentHairColorText(newHairColor);
+        }
+
+        private void BuyHairStyleOrColor(object sender, MonoGame.Extended.Input.InputListeners.MouseEventArgs e) => PurchaseHairStyleOrColor(_characterControl.RenderProperties.HairStyle, _characterControl.RenderProperties.HairColor);
+
+        private void PurchaseHairStyleOrColor(int hairStyle, int hairColor)
+        {
+            int totalCost = CalculateCost();
+            int currentGold = _characterInventoryProvider.ItemInventory.SingleOrNone(i => i.ItemID == 1)
+                                         .Map(i => i.Amount)
+                                         .ValueOr(0);
+
+            if (currentGold >= totalCost)
+            {
+                ShowConfirmationDialog(hairStyle, hairColor, totalCost);
+            }
+            else
+            {
+                ShowInsufficientFundsDialog();
+            }
+        }
+
+        private void ShowConfirmationDialog(int hairStyle, int hairColor, int cost)
+        {
+            var message = $"{_localizedStringFinder.GetString(EOResourceID.DIALOG_BARBER_DO_YOU_WANT_TO_BUY_A_NEW_HAIRSTYLE)}, {cost} {_eifFileProvider.EIFFile[1].Name}";
+            var title = _localizedStringFinder.GetString(EOResourceID.DIALOG_BARBER_BUY_HAIRSTYLE);
+            var msgBox = _messageBoxFactory.CreateMessageBox(message, title, EODialogButtons.OkCancel);
+
+            msgBox.DialogClosing += (_, e) =>
+            {
+                if (e.Result == XNADialogResult.OK)
+                {
+                    _barberActions.Purchase(hairStyle, hairColor);
+                    foreach (var notifier in _soundNotifiers)
```

Comment:
Notifiers are generally used for signalling from `EOLib` up to `EndlessClient` that something happened (because I wanted to re-invent event handlers, I guess, I regret the design decision but I digress)

You can just inject `ISfxPlayer` and play `SoundEffectID.BuySell`. See jukebox dialog.